{
  "url": "https://api.github.com/repos/facebook/react/issues/16594",
  "repository_url": "https://api.github.com/repos/facebook/react",
  "labels_url": "https://api.github.com/repos/facebook/react/issues/16594/labels{/name}",
  "comments_url": "https://api.github.com/repos/facebook/react/issues/16594/comments",
  "events_url": "https://api.github.com/repos/facebook/react/issues/16594/events",
  "html_url": "https://github.com/facebook/react/issues/16594",
  "id": 486230388,
  "node_id": "MDU6SXNzdWU0ODYyMzAzODg=",
  "number": 16594,
  "title": "useObserver and ESlint conflict",
  "user": {
    "login": "igorkamyshev",
    "id": 26767722,
    "node_id": "MDQ6VXNlcjI2NzY3NzIy",
    "avatar_url": "https://avatars3.githubusercontent.com/u/26767722?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/igorkamyshev",
    "html_url": "https://github.com/igorkamyshev",
    "followers_url": "https://api.github.com/users/igorkamyshev/followers",
    "following_url": "https://api.github.com/users/igorkamyshev/following{/other_user}",
    "gists_url": "https://api.github.com/users/igorkamyshev/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/igorkamyshev/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/igorkamyshev/subscriptions",
    "organizations_url": "https://api.github.com/users/igorkamyshev/orgs",
    "repos_url": "https://api.github.com/users/igorkamyshev/repos",
    "events_url": "https://api.github.com/users/igorkamyshev/events{/privacy}",
    "received_events_url": "https://api.github.com/users/igorkamyshev/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 12,
  "created_at": "2019-08-28T08:25:09Z",
  "updated_at": "2019-09-02T11:25:15Z",
  "closed_at": "2019-08-28T18:47:55Z",
  "author_association": "NONE",
  "body": "Hello!\r\n\r\nI found **bug** in `eslint-plugin-react-hooks` package.\r\n\r\nNow, it throw error (`React Hook \"useState\" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.`) for the following code:\r\n\r\n```js\r\nimport { useObserver } from 'mobx-react-lite';\r\n\r\nconst Component = () =>\r\n  useObserver(() => {\r\n    const [count, setCount] = useState(0);\r\n   \r\n    const increment = useCallbacl(() => setCount(c => c + 1));\r\n   \r\n   return (\r\n     <>\r\n       <p>COUNT {count}</p>\r\n       <button onClick={increment}>INC</button>\r\n     </>\r\n   )\r\n  });\r\n```\r\n\r\nI guess, it is not correct, because `useObserver` **can** be used like this.\r\n\r\nIn normal case, this code don't throw eslint error.\r\n\r\nThis issue accur on `\"eslint-plugin-react-hooks\": \"1\"`.\r\n",
  "closed_by": {
    "login": "threepointone",
    "id": 18808,
    "node_id": "MDQ6VXNlcjE4ODA4",
    "avatar_url": "https://avatars2.githubusercontent.com/u/18808?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/threepointone",
    "html_url": "https://github.com/threepointone",
    "followers_url": "https://api.github.com/users/threepointone/followers",
    "following_url": "https://api.github.com/users/threepointone/following{/other_user}",
    "gists_url": "https://api.github.com/users/threepointone/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/threepointone/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/threepointone/subscriptions",
    "organizations_url": "https://api.github.com/users/threepointone/orgs",
    "repos_url": "https://api.github.com/users/threepointone/repos",
    "events_url": "https://api.github.com/users/threepointone/events{/privacy}",
    "received_events_url": "https://api.github.com/users/threepointone/received_events",
    "type": "User",
    "site_admin": false
  }
}
